e2e/cypress/integration/channel_sidebar/channel_sidebar_spec.js

```diff
@@ -0,0 +1,130 @@
+        cy.get('#headerTeamName').should('contain', teamName);
+
+        // # Click on Off Topic
+        cy.get('.SidebarChannel:contains(Off-Topic)').should('be.visible').click();
+
+        // * Verify that the channel changed
+        cy.url().should('include', `/${teamName}/channels/off-topic`);
+        cy.get('#channelHeaderTitle').should('contain', 'Off-Topic');
+
+        // # Click on Town Square
+        cy.get('.SidebarChannel:contains(Town Square)').should('be.visible').click();
+
+        // * Verify that the channel changed
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+        cy.get('#channelHeaderTitle').should('contain', 'Town Square');
+    });
+
+    it('should mark channel as read and unread in sidebar', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team
+        cy.get('#headerTeamName').should('contain', teamName);
+
+        // * Verify that both Off Topic and Town Square are read
+        cy.get('.SidebarChannel:not(.unread):contains(Off-Topic)').should('be.visible');
+        cy.get('.SidebarChannel:not(.unread):contains(Town Square)').should('be.visible');
+
+        // # Have another user post in the Off Topic channel
+        cy.apiGetChannelByName(teamName, 'off-topic').then((response) => {
+            expect(response.status).to.equal(200);
+
+            const channel = response.body;
+            cy.postMessageAs({sender: sysadmin, message: 'Test', channelId: channel.id});
+        });
+
+        // * Verify that Off Topic is unread and Town Square is read
+        cy.get('.SidebarChannel.unread:contains(Off-Topic)').should('be.visible');
+        cy.get('.SidebarChannel:not(.unread):contains(Town Square)').should('be.visible');
+    });
+
+    it('should remove channel from sidebar after leaving it', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team
+        cy.get('#headerTeamName').should('contain', teamName);
+
+        // # Switch to Off Topic
+        cy.visit(`/${teamName}/channels/off-topic`);
+
+        // # Wait for the channel to change
+        cy.get('#channelHeaderTitle').should('contain', 'Off-Topic');
+
+        // # Click on the channel menu and select Leave Channel
+        cy.get('#channelHeaderTitle').click();
+        cy.get('#channelLeaveChannel').click();
+
+        // * Verify that we've switched to Town Square
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+        cy.get('#channelHeaderTitle').should('contain', 'Town Square');
+
+        // * Verify that Off Topic has disappeared from the sidebar
+        cy.get('.SidebarChannel:contains(Off-Topic)').should('not.exist');
```

Comment:
This failed on me.  After leaving off-topic channel, it remained on the sidebar.
![Screen Shot 2020-03-18 at 4 48 49 AM](https://user-images.githubusercontent.com/5334504/76900189-d63e0b80-68d3-11ea-9eed-d0a7739f00eb.png)

Reply:
Are you on the correct version of the web app? There's a lot of weird stuff going on with the theme in the Team Sidebar and with the status icon

Reply:
Oh, I might be behind it. Update makes all the tests passed.  Thanks